pandas/core/reshape/pivot.py

```diff
@@ -409,6 +410,17 @@ def _all_key(key):
 
         # slight hack
         new_order = [len(cols)] + list(range(len(cols)))
+
+        # GH26568 reorder_levels improperly uses a key that is also a number
+        # If a number in new_order is also a key,
+        # convert new_order to their respective keys in row_margin.index.names
+        for key in row_margin.index.names:
```

Comment:
Can't this just be a `take(new_order)` in the index names?

Reply:
@mroeschke I tried using `take()` to do what we're doing here but it doesn't seem like its doing the same thing. Could you elaborate on what `take(new_order)` should be doing here?

Reply:
If I am interpreting your for loop correctly, you're converting `new_order` which originally represented positions to labels? This should be equivalent to a `take` operation if so: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.take.html

Reply:
I'm converting positions in the `MultiIndex.names` attribute into its corresponding labels. However, using `take` on a `MultiIndex` (rather than `Index`) doesn't replicate the same functionality: 
```
(Pdb) p row_margin.index
MultiIndex([('x', 2),
            ('y', 2),
            ('z', 2)],
           names=[1, None])
(Pdb) p row_margin.index.take([1,0])
MultiIndex([('y', 2),
            ('x', 2)],
           names=[1, None])
(Pdb) p row_margin.index.take([1,0], axis=1)
MultiIndex([('y', 2),
            ('x', 2)],
           names=[1, None])
```
What I want is something like this: 
```
MultiIndex([(2, 'x'),
            (2, 'y'),
            (2, 'z')],
           names=[None, 1])
```
Are we thinking about it the same way?

Reply:
Based on your for loop though, it appears you want to take on the _names_ not the actual values?